test/e2e/service.go

```diff
@@ -797,6 +805,29 @@ var _ = Describe("Services", func() {
 	})
 })
 
+// updateService fetches a service, calls the update function on it,
+// and then attempts to send the updated service. It retries up to 2
+// times in the face of timeouts and conflicts.
+func updateService(c *client.Client, namespace, serviceName string, update func(*api.Service)) (*api.Service, error) {
+	var service *api.Service
+	var err error
+	for i := 0; i < 3; i++ {
```

Comment:
The main thing this is guarding against is conflict, which is fixed by re-fetching the service and changing the updated one. In the case of timeout, it waits on the call to actually timeout.